arch/arm64/core/reset.S

```diff
@@ -72,6 +72,20 @@ SECTION_SUBSEC_FUNC(TEXT,_reset_section,__reset_prep_c)
 out:
 	isb
 
+#if !defined(CONFIG_ARMV8_A_NS)
+	/*
+	 * When booting from EL3 we can set SP_EL1 using the sp_el1 register
+	 */
+	msr     sp_el1, x24
+#else
+	/*
+	 * When booting from EL1 to set SP_EL1 we need to change SP with SPSel
+	 * and then override SP
+	 */
+	msr     SPSel, #1
+	mov     sp, x24
```

Comment:
I am not sure if the setting sp operations can be moved before `b out`?
I mean, since `msr sp_el1, x24` is for EL3 and `msr SPSel, #1` is for EL1, is it possible to move them under the switch_el logic, like `3: ... msr sp_el1, x24 b out`, `1: ... msr SPSel, #1 mov sp, x24  out:`. So that we could get rid of the `#if defined()`?

Reply:
Yes, we can (tm)(R)